dodo.py

```diff
@@ -1073,23 +1063,38 @@ class P:
     PRETTIER_RC = ROOT / ".prettierrc"
 
 
+def _js_version_to_py_version(js_version):
+    return (
+        js_version.replace("-alpha.", "a").replace("-beta.", "b").replace("-rc.", "rc")
+    )
+
+
 class D:
     # data
     APP = json.loads(P.APP_PACKAGE_JSON.read_text(**C.ENC))
     APP_VERSION = APP["version"]
     APPS = APP["jupyterlite"]["apps"]
+    APP_SCHEMA = json.loads(P.APP_SCHEMA.read_text(**C.ENC))
+    APP_SCHEMA_DEFS = APP_SCHEMA["definitions"]
+    APP_SCHEMA_PYOLITE = APP_SCHEMA_DEFS["pyolite-settings"]
 
     # derive the PEP-compatible version
-    PY_VERSION = (
-        APP["version"]
-        .replace("-alpha.", "a")
-        .replace("-beta.", "b")
-        .replace("-rc.", "rc")
-    )
+    PY_VERSION = _js_version_to_py_version(APP["version"])
 
     PACKAGE_JSONS = {
-        p.parent.name: json.loads(p.read_text(**C.ENC)) for p in P.PACKAGE_JSONS
+        parent: json.loads(p.read_text(**C.ENC))
+        for parent, p in P.PACKAGE_JSONS.items()
     }
+    PYODIDE_JS_VERSION = PACKAGE_JSONS["pyolite-kernel"]["devDependencies"]["pyodide"]
+    PYODIDE_CDN_URL = APP_SCHEMA_PYOLITE["properties"]["pyodideUrl"]["default"]
+    PYODIDE_VERSION = _js_version_to_py_version(PYODIDE_JS_VERSION)
+    PYODIDE_JS = PYODIDE_CDN_URL.split("/")[-1]
+    PYODIDE_ARCHIVE = f"pyodide-{PYODIDE_VERSION}.tar.bz2"
```

Comment:
this name will need to change, removing `-build` in #909

Reply:
Sounds good let's keep track of that in #909.